### PR TITLE
PBLAS library update

### PR DESCRIPTION
**Pull request by alexnick83**
This PR updates the PBLAS (DaCe) library so that programs utilizing it work properly with the newest DaCe version. It also updates the Intel ScaLAPACK environment to work with the newest versions of MKL. Support for (non-Intel) ScaLAPACK will also be added.

### CODE REVIEW

**Review by tbennun — APPROVED** [review]
wait, GitHub UI bug

File: dace/libraries/pblas/environments/intel_mkl_mpich.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        if os.path.isfile(simd_libfile):
+            libfiles.append(simd_libfile)
+
+        return libfiles
+
+    @staticmethod
+    def cmake_link_flags():
+
+        # From https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse
+        def str2bool(v: Union[None, str, bool]) -> bool:
```
Comment (tbennun) [inline dace/libraries/pblas/environments/intel_mkl_mpich.py:132]: https://github.com/spcl/dace/blob/master/dace/config.py#L45

File: dace/libraries/pblas/environments/intel_mkl_mpich.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            if isinstance(v, bool):
+                return v
+            if isinstance(v, str):
+                if v.lower() in ('yes', 'true', 't', 'y', '1'):
+                    return True
+                if v.lower() in ('no', 'false', 'f', 'n', '0'):
+                    return False
+            return False
+
+        is_daint = str2bool(os.getenv('IS_DAINT'))
```
Comment (tbennun) [inline dace/libraries/pblas/environments/intel_mkl_mpich.py:144]: oh no

File: dace/libraries/pblas/environments/intel_mkl_mpich.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        if is_daint:
+            mpichlib_path = "-L /opt/cray/pe/mpt/7.7.18/gni/mpich-gnu/8.2/lib"
```
Comment (tbennun) [inline dace/libraries/pblas/environments/intel_mkl_mpich.py:146]: that is very bad

File: samples/distributed/polybench.py [hunk 4]
```diff
@@ -0,0 +1,1293 @@
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
+""" Assortment of distributed Polybench kernels. """
```
Comment (tbennun) [inline samples/distributed/polybench.py:2]: does this qualify as a sample? this is quite long

Comment (tbennun) [reply samples/distributed/polybench.py:2]: maybe a benchmark in a new folder called `benchmarks/`, but not a sample to me

**Review by tbennun — CHANGES_REQUESTED** [review]
amending to comment on the IS_DAINT issue

File: dace/libraries/pblas/environments/intel_mkl_mpich.py (hunk 2 above)
Comment (alexnick83) [reply dace/libraries/pblas/environments/intel_mkl_mpich.py:144]: We are keeping this as an example for adjusting the environment to work on different systems. It is impossible to work around this with the current environment API in an elegant way.

File: samples/distributed/polybench.py (hunk 4 above)
Comment (alexnick83) [reply samples/distributed/polybench.py:2]: It is *many* samples, but it is better to group them together to work around `MPI_Finalize`.

**Review by tbennun — APPROVED** [review]
LGTM

**Pull request merged**